plover/gui_qt/machine_options.py

```diff
@@ -63,10 +67,13 @@ def _update(self, field, value):
 
     def on_scan(self):
         self.port.clear()
-        self.port.addItems(sorted(f"{port.device} ({port.description})" for port in comports()))
+        for port in comports():
+            portItem = QStandardItem(str(port))
+            portItem.setData(port.device)
+            self.ports.appendRow(portItem)
 
     def on_port_changed(self, value):
-        port = value.split(" ")[0]
+        port = self.ports.item(self.port.currentIndex()).data()
```

Comment:
This doesn't work if I just press the "scan" button below... (`item()` returns None)

Reply:
This happens when you don't select something (e.g. on start up). I did a check on the item but there's a case for putting the check on `currentIndex() == -1` instead. 9819ed0eb54681b1eb9fd5432a867b79c5d2e69b